ocaml/xapi/xapi_auth.ml

```diff
@@ -49,5 +49,12 @@ let get_group_membership ~__context ~subject_identifier =
 
 let get_subject_information_from_identifier ~__context ~subject_identifier =
   call_with_exception_handler (fun () ->
-      (Ext_auth.d ()).query_subject_information ~__context subject_identifier
+      try
+        (* Query from xapi db first *)
+        Xapi_subject.query_subject_information_from_db ~__context
+          subject_identifier
+      with Auth_signature.Subject_cannot_be_resolved ->
+        (* Not found, fall back to query AD *)
+        Xapi_subject.query_subject_information_from_AD ~__context
+          subject_identifier
```

Comment:
I think it makes sense to abstract away the source that is queried to fetch the subject:

```suggestion
Xapi_subject.query_subject_information ~__context subject_identifier
```
And on xapi_subject only expose a single function:
```ocaml
let query_subject_information ~__context identifier =
      try
        (* Query from xapi db first *)
        query_subject_information_from_db ~__context subject_identifier
      with Auth_signature.Subject_cannot_be_resolved ->
        (* Not found, fall back to query AD *)
        query_subject_information_from_AD ~__context subject_identifier
```

Also how does this interact with the cache? Is the information available opn the db even without cache?

Reply:
Yes, This information is available in DB even without cache.